driver/main.c

```diff
@@ -1284,11 +1284,14 @@ static const unsigned char compat_nas[21] = {
 #ifdef _HAS_SOCKETCALL
 static enum ppm_event_type parse_socketcall(struct event_filler_arguments *filler_args, struct pt_regs *regs)
 {
-	unsigned long __user args[2];
+	unsigned long __user args[6] = {};
 	unsigned long __user *scargs;
 	int socketcall_id;
-
-	syscall_get_arguments(current, regs, 0, 2, args);
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 1, 0))
```

Comment:
Just a thought, but maybe it'd be nicer to do:

```c
#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 1, 0))
#define SYSCALL_GET_ARGUMENTS(current, regs, args) syscall_get_arguments((current), (regs), 0, 6, (args))
#else 
#define SYSCALL_GET_ARGUMENTS(current, regs, args) syscall_get_arguments((current), (regs), (args))
#endif 
```

And then, for each call to the API, just make it:

```c
SYSCALL_GET_ARGUMENTS(current, regs, args);
```

That way you avoid duplicating the logic at all the call sites.